hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/KafkaSource.java

```diff
@@ -76,4 +86,14 @@ public void onCommit(String lastCkptStr) {
       offsetGen.commitOffsetToKafka(lastCkptStr);
     }
   }
+
+  protected AvroConvertor getAvroConverter(Boolean preProcessedSchema) {
+    if (schemaProvider == null) {
+      throw new HoodieException("Needed schema provider for avro deserializer");
+    }
+    if (preProcessedSchema && (schemaProvider instanceof SchemaProviderWithPostProcessorDisableSource)) {
+      return new AvroConvertor(((SchemaProviderWithPostProcessorDisableSource) schemaProvider).getSourceSchemaWithoutPostProcessor());
```

Comment:
this is what I was pointing out earlier. We should call a generic method. internally whether post processor for source is enabled/disabled should not matter to callers.